### PR TITLE
Add the SolidButton component

### PR DESCRIPTION
**Pull request by BelfDev**
# SolidButton component

This component is built on the top of the `Button` component. The implementation is showcased below:

**Normal state**
Screen Shot 2020-03-02 at 12 24 13 PM

**Focused or Hovered states**
Screen Shot 2020-03-02 at 12 24 21 PM

<hr>
More details about this PR:

- Create the `SolidButton` component
- Create a `resources/styles` directory with `_colors.scss` and `_typography.scss` [partial Sass files](https://sass-lang.com/documentation/syntax) to handle the project Colors and Typography styles.
- Establish a color naming convention based on [Material Design](https://material.io/design/color/the-color-system.html#color-theme-creation) guidelines.
- Apply the shadow animation based on Material Design [elevation rules](https://material.io/design/environment/elevation.html#).

_Note: I have added a new task on [Trello](https://trello.com/b/RbCv3E2D/internal-tech) to integrate the project with [Typography.js](https://www.gatsbyjs.org/docs/typography-js/). This dependency will help us manage our custom font theming throughout the project. Therefore, **the SolidButton component does not contain the correct typography styling yet**._

### CODE REVIEW

File: src/components/common/buttons/SolidButton/style.module.scss [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  border-radius: 163px;
+  font: Semibold 18px/23px $typographic-font-family;
+  color: $typography-on-surface-light;
+  box-shadow: 0 2px 6px $color-shadow;
+  transition: box-shadow .3s;
+  text-align: center;
+  text-transform: uppercase;
+
+  &:hover,
+  &:focus {
```
Comment (yagotome) [inline src/components/common/buttons/SolidButton/style.module.scss]: Why focused button would equal to hovered? It seems the button is kind of selected after you have touched (don't go back automaticly to normal state).

```suggestion
  &:hover {
```

File: src/resources/styles/_typography.scss [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Base
+$typographic-font-family: Source Sans Pro, sans-serif;
```
Comment (yagotome) [inline src/resources/styles/_typography.scss]: Great! :+1:

File: src/components/common/buttons/SolidButton/SolidButton.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import styles from './style.module.scss';
+import Button from '../Button';
+
+const SolidButton = ({ className, title }) => {
+  return (
+    <Button className={classNames(styles.solidButton, className)}>
+      {title}
```
[the remark this answers is not in the document]
Comment (BelfDev) [reply src/components/common/buttons/SolidButton/SolidButton.js]: I'm focusing on the HomePage first section. We'll evolve the components continuously.

File: src/components/common/buttons/SolidButton/SolidButton.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    </Button>
+  );
+};
+
+SolidButton.propTypes = {
+  className: PropTypes.string,
+};
```
[the remark this answers is not in the document]
Comment (BelfDev) [reply src/components/common/buttons/SolidButton/SolidButton.js]: Thanks! It's [fixed](https://github.com/lutick/website/pull/10/commits/4f5d4295dc61ed4505a4b210f4f0de8876338908).

File: src/components/common/buttons/SolidButton/style.module.scss (hunk 1 above)
Comment (BelfDev) [reply src/components/common/buttons/SolidButton/style.module.scss]: You are right. [Fixed](https://github.com/lutick/website/pull/10/commits/442faaea79cbf57fe1595f1922fed694e19b89a2).

File: src/resources/styles/_typography.scss (hunk 2 above)
Comment (BelfDev) [reply src/resources/styles/_typography.scss]: I'm glad you liked it! But we'll probably move to [Typography.js](https://www.gatsbyjs.org/docs/typography-js/).

**Pull request merged**